### PR TITLE
[WIP] Add DHCP support

### LINKED ISSUE #56: DHCPv6
DHCPv6 will be nice for IPv6 only deployments.

While IPv6 can utilize RA for stateless configuration, we need to get a hostname for the TLS certificate.

### PR DESCRIPTION
**Pull request by bluecmd**
Fixes #56.

### CODE REVIEW

File: pkg/bmc/interface.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"time"
+
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+	"github.com/insomniacslk/dhcp/dhcpv6"
```
Comment (hugelgupf) [inline pkg/bmc/interface.go]: Did you look at u-root's dhcp6client?

Comment (bluecmd) [reply pkg/bmc/interface.go]: I did not! I assumed u-root didn't have a dhcp6client. I'll compare them, but it looks pretty nice from initial scan. Thanks for flagging.

**Comment by insomniacslk** [comment]
@bluecmd as discussed privately, better to switch to u-root's library

**Comment by bluecmd** [comment]
@hugelgupf I'm using u-root's DHCPv4 library now, and I'm not really understanding why it's acting up on me.

This is the console output:
```
1970/01/01 00:01:50 DEBUG: client.Request
[..]
1970/01/01 00:02:21 DEBUG: sleep for 1m0s, error on "eth0": context deadline exceeded
```

So it believes it never got any reply. However tshark says:

```
Frame 2542: 290 bytes on wire (2320 bits), 290 bytes captured (2320 bits) on interface 0
    Interface id: 0 (ens224)
        Interface name: ens224
    Encapsulation type: Ethernet (1)
    Arrival Time: Sep 30, 2018 16:50:40.014935896 CEST
    [Time shift for this packet: 0.000000000 seconds]
    Epoch Time: 1538319040.014935896 seconds
    [Time delta from previous captured frame: 0.175214850 seconds]
    [Time delta from previous displayed frame: 10.019534168 seconds]
    [Time since reference or first frame: 243.476442177 seconds]
    Frame Number: 2542
    Frame Length: 290 bytes (2320 bits)
    Capture Length: 290 bytes (2320 bits)
    [Frame is marked: False]
    [Frame is ignored: False]
    [Protocols in frame: eth:ethertype:ip:udp:bootp]
Ethernet II, Src: b2:60:ba:39:c3:54 (b2:60:ba:39:c3:54), Dst: Broadcast (ff:ff:ff:ff:ff:ff)
    Destination: Broadcast (ff:ff:ff:ff:ff:ff)
        Address: Broadcast (ff:ff:ff:ff:ff:ff)
        .... ..1. .... .... .... .... = LG bit: Locally administered address (this is NOT the factory default)
        .... ...1 .... .... .... .... = IG bit: Group address (multicast/broadcast)
    Source: b2:60:ba:39:c3:54 (b2:60:ba:39:c3:54)
        Address: b2:60:ba:39:c3:54 (b2:60:ba:39:c3:54)
        .... ..1. .... .... .... .... = LG bit: Locally administered address (this is NOT the factory default)
        .... ...0 .... .... .... .... = IG bit: Individual address (unicast)
    Type: IPv4 (0x0800)
Internet Protocol Version 4, Src: 0.0.0.0, Dst: 255.255.255.255
    0100 .... = Version: 4
    .... 0101 = Header Length: 20 bytes (5)
    Differentiated Services Field: 0x00 (DSCP: CS0, ECN: Not-ECT)
        0000 00.. = Differentiated Services Codepoint: Default (0)
        .... ..00 = Explicit Congestion Notification: Not ECN-Capable Transport (0)
    Total Length: 276
    Identification: 0x0000 (0)
    Flags: 0x0000
        0... .... .... .... = Reserved bit: Not set
        .0.. .... .... .... = Don't fragment: Not set
        ..0. .... .... .... = More fragments: Not set
        ...0 0000 0000 0000 = Fragment offset: 0
    Time to live: 30
    Protocol: UDP (17)
    Header checksum: 0x9bda [validation disabled]
    [Header checksum status: Unverified]
    Source: 0.0.0.0
    Destination: 255.255.255.255
User Datagram Protocol, Src Port: 68, Dst Port: 67
    Source Port: 68
    Destination Port: 67
    Length: 256
    Checksum: 0xecc8 [unverified]
    [Checksum Status: Unverified]
    [Stream index: 2]
Bootstrap Protocol (Discover)
    Message type: Boot Request (1)
    Hardware type: Ethernet (0x01)
    Hardware address length: 6
    Hops: 0
    Transaction ID: 0x5b813208
    Seconds elapsed: 0
    Bootp flags: 0x8000, Broadcast flag (Broadcast)
        1... .... .... .... = Broadcast flag: Broadcast
        .000 0000 0000 0000 = Reserved flags: 0x0000
    Client IP address: 0.0.0.0
    Your (client) IP address: 0.0.0.0
    Next server IP address: 0.0.0.0
    Relay agent IP address: 0.0.0.0
    Client MAC address: b2:60:ba:39:c3:54 (b2:60:ba:39:c3:54)
    Client hardware address padding: 00000000000000000000
    Server host name not given
    Boot file name not given
    Magic cookie: DHCP
    Option: (53) DHCP Message Type (Discover)
        Length: 1
        DHCP: Discover (1)
    Option: (57) Maximum DHCP Message Size
        Length: 2
        Maximum DHCP Message Size: 1500
    Option: (255) End
        Option End: 255

Frame 2543: 330 bytes on wire (2640 bits), 330 bytes captured (2640 bits) on interface 0
    Interface id: 0 (ens224)
        Interface name: ens224
    Encapsulation type: Ethernet (1)
    Arrival Time: Sep 30, 2018 16:50:40.015393460 CEST
    [Time shift for this packet: 0.000000000 seconds]
    Epoch Time: 1538319040.015393460 seconds
    [Time delta from previous captured frame: 0.000457564 seconds]
    [Time delta from previous displayed frame: 0.000457564 seconds]
    [Time since reference or first frame: 243.476899741 seconds]
    Frame Number: 2543
    Frame Length: 330 bytes (2640 bits)
    Capture Length: 330 bytes (2640 bits)
    [Frame is marked: False]
    [Frame is ignored: False]
    [Protocols in frame: eth:ethertype:ip:udp:bootp]
Ethernet II, Src: Vmware_b7:8c:60 (00:50:56:b7:8c:60), Dst: b2:60:ba:39:c3:54 (b2:60:ba:39:c3:54)
    Destination: b2:60:ba:39:c3:54 (b2:60:ba:39:c3:54)
        Address: b2:60:ba:39:c3:54 (b2:60:ba:39:c3:54)
        .... ..1. .... .... .... .... = LG bit: Locally administered address (this is NOT the factory default)
        .... ...0 .... .... .... .... = IG bit: Individual address (unicast)
    Source: Vmware_b7:8c:60 (00:50:56:b7:8c:60)
        Address: Vmware_b7:8c:60 (00:50:56:b7:8c:60)
        .... ..0. .... .... .... .... = LG bit: Globally unique address (factory default)
        .... ...0 .... .... .... .... = IG bit: Individual address (unicast)
    Type: IPv4 (0x0800)
    Trailer: ffffb87cb12b838f1d00
    Frame check sequence: 0x1ea3b12b incorrect, should be 0xd5113533
        [Expert Info (Error/Checksum): Bad checksum [should be 0xd5113533]]
            [Bad checksum [should be 0xd5113533]]
            [Severity level: Error]
            [Group: Checksum]
    [FCS Status: Bad]
Internet Protocol Version 4, Src: 10.0.10.1, Dst: 255.255.255.255
    0100 .... = Version: 4
    .... 0101 = Header Length: 20 bytes (5)
    Differentiated Services Field: 0x10 (DSCP: Unknown, ECN: Not-ECT)
        0001 00.. = Differentiated Services Codepoint: Unknown (4)
        .... ..00 = Explicit Congestion Notification: Not ECN-Capable Transport (0)
    Total Length: 302
    Identification: 0x0000 (0)
    Flags: 0x4000, Don't fragment
        0... .... .... .... = Reserved bit: Not set
        .1.. .... .... .... = Don't fragment: Set
        ..0. .... .... .... = More fragments: Not set
        ...0 0000 0000 0000 = Fragment offset: 0
    Time to live: 128
    Protocol: UDP (17)
    Header checksum: 0xe5ae [validation disabled]
    [Header checksum status: Unverified]
    Source: 10.0.10.1
    Destination: 255.255.255.255
User Datagram Protocol, Src Port: 67, Dst Port: 68
    Source Port: 67
    Destination Port: 68
    Length: 282
    Checksum: 0x172a [unverified]
    [Checksum Status: Unverified]
    [Stream index: 3]
Bootstrap Protocol (Offer)
    Message type: Boot Reply (2)
    Hardware type: Ethernet (0x01)
    Hardware address length: 6
    Hops: 0
    Transaction ID: 0x5b813208
    Seconds elapsed: 0
    Bootp flags: 0x8000, Broadcast flag (Broadcast)
        1... .... .... .... = Broadcast flag: Broadcast
        .000 0000 0000 0000 = Reserved flags: 0x0000
    Client IP address: 0.0.0.0
    Your (client) IP address: 10.0.10.155
    Next server IP address: 10.0.10.1
    Relay agent IP address: 0.0.0.0
    Client MAC address: b2:60:ba:39:c3:54 (b2:60:ba:39:c3:54)
    Client hardware address padding: 00000000000000000000
    Server host name not given
    Boot file name not given
    Magic cookie: DHCP
    Option: (1) Subnet Mask
        Length: 4
        Subnet Mask: 255.255.255.0
    Option: (3) Router
        Length: 4
        Router: 10.0.10.1
    Option: (6) Domain Name Server
        Length: 4
        Domain Name Server: 8.8.8.8
    Option: (51) IP Address Lease Time
        Length: 4
        IP Address Lease Time: (4000s) 1 hour, 6 minutes, 40 seconds
    Option: (53) DHCP Message Type (Offer)
        Length: 1
        DHCP: Offer (2)
    Option: (54) DHCP Server Identifier
        Length: 4
        DHCP Server Identifier: 10.0.10.1
    Option: (255) End
        Option End: 255
```

I.e. the DHCP server certainly send the request.

**Pull request closed without merging**